File: 2021/10/06/GoatLatin.java
/**
 * https://leetcode.com/problems/goat-latin/
 * @author Andrew Jarombek
 * @since 10/6/2021
 */

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

public class GoatLatin {
    public String toGoatLatin(String sentence) {
        Set<Character> vowels = new HashSet<>(Arrays.asList('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'));
        StringBuilder sb = new StringBuilder();

        int index = 1;
        for (String s : sentence.split(" ")) {
            if (sb.length() > 0) {
                sb.append(' ');
            }

            if (vowels.contains(s.charAt(0))) {
                sb.append(s);
            } else {
                sb.append(s.substring(1));
                sb.append(s.charAt(0));
            }

            sb.append("ma");

            for (int i = 0; i < index; i++) {
                sb.append('a');
            }

            index++;
        }

        return sb.toString();
    }
}

File: 2022/04/30/goatLatin.ts
/**
 * https://leetcode.com/problems/goat-latin/
 * @author Andrew Jarombek
 * @since 4/30/2022
 */

function toGoatLatin(sentence: string): string {
    let vowels = new Set<string>(['A', 'E', 'I', 'O', 'U', 'a', 'e', 'i', 'o', 'u']);
    let words = sentence.split(" ");

    for (let i = 0; i < words.length; i++) {
        let chars = words[i].split("");
        let start = 0;

        if (!vowels.has(chars[0])) {
            start = 1;
            chars.push(chars[0]);
        }

        chars.push("ma");

        for (let index = 0; index <= i; index++)
            chars.push("a");

        words[i] = chars.join("").substring(start);
    }

    return words.join(" ");
}